### PR TITLE
Allow the `ui_menu` action to trigger context menus in Tree

### PR DESCRIPTION
**Pull request by eswartz**
My right mouse button is dying. Not gonna replace it yet. :)

This follows the pattern seen in other GUI nodes.

### CODE REVIEW

**Review by Mickeon — COMMENTED** [review]
Needs a simple test to make sure it actually works in practice

File: scene/gui/tree.cpp [hunk 1]
```diff
@@ -3534,6 +3534,10 @@ void Tree::gui_input(const Ref<InputEvent> &p_event) {
 			}
 		}
 		accept_event();
+	} else if (p_event->is_action("ui_menu") && p_event->is_pressed()) {
+		// Show the context menu
```
[the remark this answers is not in the document]
Comment (eswartz) [reply scene/gui/tree.cpp]: Added a test and fixed the comment.

**Comment by eswartz** [comment]
> Needs a simple test to make sure it actually works in practice

Added a test (a lonely test in a new file, lol).

**Comment by Mickeon** [comment]
That's not what I meant at all but it's incredible you took it that way. If Tree did not have tests before this may be a good opportunity to start having them.

**Comment by KoBeWi** [comment]
That looks like hack. The Tree assumes that menu is spawned on right-click (while right-click might do something else) and will also spawn it wherever the cursor is at the moment, which can be totally somewhere else than the Tree.

The only proper solution I can think of is adding a new signal `open_context_menu`, which is emitted on right click and on `ui_menu`. When emitted from keyboard, the popup should spawn at the selected item. idk if there is better solution, all of the Controls that handle `ui_menu` are also responsible for spawning the menu, which is why the handler wasn't added to all classes.

Another thing is that some Trees might have `ui_menu` handled externally.

**Comment by eswartz** [comment]
@KoBeWi You're right, this might be too quick-n-dirty. It works in the project editor and scene tree, which is where I needed it once my mouse started to die. Given my use case, obviously I am already using my mouse to position the menu in order to select the thing I need the menu on. :) But I did see the behavior you mentioned after some experiments to move focus somewhere else.

I can just keep this as a local patch for now. I'm not confident enough to tackle adding API for this.

**Pull request closed without merging**